taxonomy-editor-frontend/src/pages/editentry/index.jsx

```diff
@@ -0,0 +1,39 @@
+import { Typography } from "@mui/material";
+import { useParams } from "react-router-dom";
+import useFetch from "../../components/useFetch";
+import { API_URL } from "../../constants";
+import AccumulateAllComponents from "./AccumulateAllComponents";
+
+const EditEntry = () => {
+    let url = API_URL;
+    let isEntry = false;
+    const { id } = useParams();
+    if (id.startsWith('__header__')) { url += 'header/' }
+    else if (id.startsWith('__footer__')) { url += 'footer/' }
+    else if (id.startsWith('synonym')) { url += `synonym/${id}/` }
+    else if (id.startsWith('stopword')) { url += `stopword/${id}/` }
+    else { url += `entry/${id}/`; isEntry = true; }

```

Comment:
what about passing this logic to a hook or even just a function? That way you could call it where needed. In any case it may be better to at least pass this piece of logic to the `AccumulateAllComponents` as is not really used in here

Reply:
This logic is being used to find the url to which, the request has to be sent. My goal was to first retrieve the information in `editentry/index.jsx` and pass along the information to `AccumulateAllComponents.jsx`. I would be able to bring this logic to `AccumulateAllComponents` but I thought it would make the component even more cluttered and difficult to understand.

Regarding making it a function/hook, this logic will only be used here and will most probably not be reused anywhere else. @nobeeakon do you think there is any other way to structure it, without compromising the readability?

Reply:
if not reused anywhere else think is ok to not doing a hook/function. I'd advice anyways to do it in the `AccumulateAllComponents` as that is the one using it, besides you need to pass the `url` to the `FetchRelations` so if you fetch the data at the `index` level you still need to have the `url` in there